File: src/main/java/com/example/demoimdb/model/Account.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "account")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "username")
    private String username;
    @Column(name = "password")
    private String password;
    @Column(name = "email")
    private String email;
    @Column(name = "role")
    private String role;
    @JsonIgnore
    @OneToMany(mappedBy = "account")
    private List<Comment> listComments;
    @JsonIgnore
    @OneToMany(mappedBy = "account")
    private List<Rating> listRatings;
}

File: src/entities/Account.ts
import { PrimaryGeneratedColumn, Column, Entity, OneToMany } from 'typeorm';
import { Rating } from './Rating';
import { Comment } from './Comment';

@Entity({ name: 'accounts' })
export class Account {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  username: string;

  @Column()
  password: string;

  @Column()
  email: string;

  @Column()
  role: number;

  @OneToMany(() => Rating, (rating) => rating.account)
  ratings: Rating[];

  @OneToMany(() => Comment, (comment) => comment.account)
  comments: Comment[];
}
